### ISSUE TITLE
twistd   twisted daemon script fails to run, because of '_twistd_unix.py'

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8155]
|<img alt="bhyz8080's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| bhyz8080 reported|
|-|-|
|Trac ID|trac#8155|
|Type|defect|
|Created|2015-12-24 05:07:11Z|
|Branch|https://github.com/twisted/twisted/tree/twistdpy3-fix-8155|

twisted calls _twistd_unix.py (running Python 3.5.1 by Anaconda)

line 92 has syntax error:

  is: 
      if why[0] = errno.ESRCH

  should be:
      if why.args[0] = errno.ESRCH

<details><summary>Searchable metadata</summary>

```
trac-id__8155 8155
type__defect defect
reporter__bhyz8080 bhyz8080
priority__normal normal
milestone__None None
branch__branches_twistdpy3_fix_8155 branches/twistdpy3-fix-8155
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__twistd twistd
time__1450933631639297 1450933631639297
changetime__1474198049187875 1474198049187875
version__None None
owner__Mark_Williams__markrwilliams_____ Mark Williams <markrwilliams@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [46601]) Branching to twistdpy3-fix-8155.

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_2"></a>|@markrwilliams commented|
|-|-|

I'm sure you'll find this when you review the file, but just in case:

``` UnixApplicationRunner.postApplication ``` and ``` UnixApplicationRunner._waitForStart ``` need to send and check against ``` b"0" ```.

I'd be happy to help port this file/write tests.

**Comment by twisted-trac** [comment]
|[<img alt="evgeni's avatar" src="https://avatars.githubusercontent.com/u/94284?s=50" width="50" height="50">](https://github.com/evgeni)<a name="note_3"></a>|@evgeni commented|
|-|-|

the why part was fixed (slightly differently) in https://github.com/twisted/twisted/commit/95f5151da6c18f1b18ade8c73d00ceb72be853ca

**Comment by twisted-trac** [comment]
|[<img alt="evgeni's avatar" src="https://avatars.githubusercontent.com/u/94284?s=50" width="50" height="50">](https://github.com/evgeni)<a name="note_4"></a>|@evgeni commented|
|-|-|

Replying to [markrwilliams](#note_2):
> I'm sure you'll find this when you review the file, but just in case:
> 
> ``` UnixApplicationRunner.postApplication ``` and ``` UnixApplicationRunner._waitForStart ``` need to send and check against ``` b"0" ```.
> 
> I'd be happy to help port this file/write tests.

I opened a PR for that: https://github.com/twisted/twisted/pull/461
not sure how to add tests here, though.

@markrwilliams ideas?

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_6"></a>|@markrwilliams commented|
|-|-|

Replying to [evgeni](#note_4):
>
> I opened a PR for that: https://github.com/twisted/twisted/pull/461
> not sure how to add tests here, though.
> 
> @markrwilliams ideas?

Thanks so much for tackling this!  I think the test changes in the PR provide sufficient coverage for your changes.  However, I've got some minor review comments that I'll put there, so stay tuned!

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_7"></a>|@markrwilliams set owner to @markrwilliams<br>@markrwilliams set status to `closed`|
|-|-|

In changeset c413aac3888dea2202c0dc26f978d7f88b4b837a
```
#!CommitTicketReference repository="" revision="c413aac3888dea2202c0dc26f978d7f88b4b837a"
Merge evgeni/8155-evgeni-bytes-compare: use bytes not str when working with pipes

Author: evgeni
Reviewer: markrwilliams
Fixes: #8155

This ensures twisted.scripts._twistd_unix writes bytes to the pipes
used to orchestrate daemonization.  twistd daemonization should now
work on Python 3.
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_8"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
